weldx/transformations.py

```diff
@@ -379,45 +518,449 @@ def _calculate_orthogonal_axis(a_0, a_1):
+        return self._dataset.basis.transpose(..., "c", "v")
 
     @property
-    def orientation(self):
+    def orientation(self) -> xr.DataArray:
         """
         Get the coordinate systems orientation matrix.
 
         This function is identical to the 'basis' function.
 
         :return: Orientation matrix
         """
-        return self._orientation
+        return self._dataset.basis.transpose(..., "c", "v")
 
     @property
-    def origin(self):
+    def origin(self) -> xr.DataArray:
         """
         Get the coordinate systems origin.
 
         This function is identical to the 'location' function.
 
         :return: Origin of the coordinate system
         """
-        return self._location
+        return self._dataset.origin.transpose(..., "c")
 
     @property
-    def location(self):
+    def location(self) -> xr.DataArray:
         """
         Get the coordinate systems location.
 
         This function is identical to the 'origin' function.
 
         :return: Location of the coordinate system.
         """
-        return self._location
+        return self._dataset.origin.transpose(..., "c")
+
+    @property
+    def time(self) -> pd.DatetimeIndex:
+        """
+        Get the time union of the local coordinate system (or None if system is static).
+
+        :return: DateTimeIndex-like time union
+        """
+
+        if "time" in self._dataset.coords:
+            return pd.DatetimeIndex(self._dataset.time.data)
+        return None
+
+    @property
+    def dataset(self) -> xr.Dataset:
+        """
+        Get the underlying xarray Dataset.
+
+        :return: xarray Dataset with origin and basis as DataVariables.
+        """
+        return self._dataset
+
+    def interp_time(self, time: pd.DatetimeIndex) -> "LocalCoordinateSystem":
+        """
+        Interpolates the data in time.
+
+        :param time: Series of times.
+        :return: Coordinate system with interpolated data
+        """
+        if not isinstance(time, pd.DatetimeIndex):
+            raise Exception("Invalid parameter type.")
+        basis = ut.xr_interp_orientation_in_time(self.basis, time)
+        origin = ut.xr_interp_coodinates_in_time(self.origin, time)
+
+        return LocalCoordinateSystem(basis, origin)
+
+    def interp_time_like(
+        self, refernce: "LocalCoordinateSystem"
+    ) -> "LocalCoordinateSystem":
+        """
+        Interpolates the data in time using another coordinate systems time axis.
+
+        :param refernce: Coordinate system that provides the reference time.
+        :return: Coordinate system with interpolated data
+        """
+        if not isinstance(refernce, LocalCoordinateSystem):
+            raise Exception("Invalid reference type")
+        if refernce.time is not None:
+            times = refernce.time
+            return self.interp_time(times)
+        else:
+            raise Exception("Reference coordinate system has no time component")
```

Comment:
Do we need this for error handling? Maybe we can just use `lcs.interp_time(reference.time)` instead ?